sys/net/application_layer/dhcpv6/client.c

```diff
@@ -248,8 +256,11 @@ static inline size_t _compose_elapsed_time_opt(dhcpv6_opt_elapsed_time_t *time)
 }
 
 static inline size_t _compose_mud_url_opt(dhcpv6_opt_mud_url_t *mud_url_opt,
-                                          const char *mud_url, size_t len_max)
+                                          size_t len_max)
 {
+#ifndef MODULE_GNRC_DHCPV6_CLIENT_MUD_URL
+    return 0;
+#endif /* MODULE_GNRC_DHCPV6_CLIENT_MUD_URL */
```

Comment:
If that works you can also do

```suggestion
    if (!IS_USED(MODULE_GNRC_DHCPV6_CLIENT_MUD_URL)) {
        return 0;
    }
```

You'll have to remove the conditional definition of `mud_url`, but the optimizer should throw it out if it's not used.
You can check that the RAM/ROM size does not change.

Reply:
Looks good, I added it down below :)